### ISSUE TITLE
XF86AudioPlay keyboard key  ignored

### ISSUE THREAD

**Issue by phocean** [linuxmint/Cinnamon#7852]
```
 * Cinnamon 3.6.7
 * Distribution - Ubuntu 18.04
 * Graphics Intel driver
 * 64 bit
 ```

**Issue**

My headphone has a key that triggers XF86AudioPlay.

It works in Gnome with any media program (VLC, gnome-mpv, rhythmbox). When no player has been launched, an OSD is still displayed to show that it was recognized.

In Cinnamon, it works only with VLC (which I believe has its own hook).
Otherwise, it does not work in gnome-mpv, rhythmbox or anything else I have tried so far.

Pressing the key does not seem to trigger anything in the DE (even when it is attached to other events, like window managing).

Here is my current shortcut setting for this key:

![capture du 2018-09-15 12-48-37](https://user-images.githubusercontent.com/4280885/45585508-bcd69080-b8e5-11e8-8137-69e5c3f01e7d.png)

xev output:

```
KeyRelease event, serial 38, synthetic NO, window 0x6400001,
    root 0x16a, subw 0x0, time 623575, (470,170), root:(1332,625),
    state 0x0, keycode 208 (keysym 0x1008ff14, XF86AudioPlay), same_screen YES,
    XKeysymToKeycode returns keycode: 172
    XLookupString gives 0 bytes: 
    XFilterEvent returns: False
```

**Comment by mtwebster** [comment]
Is the key recognized by the shortcuts ui?  Meaning - when in teach mode, does it register?

Do you have any other media control keys (perhaps on the keyboard) - next, previous, stop - do they work?

In the screenshot you're looking at the shortcut for "Pause" - why is that?  By default XF86AudioPlay is the shortcut for "Play" (which will toggle play/pause ordinarily.)  You cannot define the same key for two shortcuts there (which should be enforced by the gui, unless you go into the settings backend and assign it manually.)

Try the following to debug this a bit further - in a terminal, kill csd-media-keys twice:
```
sudo killall csd-media-keys
sudo killall csd-media-keys
```
(it restarts itself automatically the first time)

then,
```
csd-media-keys --verbose
```
This will show interaction with grabbed keys, as well as with the MPRIS interface used to control media players.

**Comment by phocean** [comment]
>  Is the key recognized by the shortcuts ui? Meaning - when in teach mode, does it register?
 
Yes, it does register as "Démarrage Lecture" as on the screenshot.

> Do you have any other media control keys (perhaps on the keyboard) - next, previous, stop - do they work?

No, I have not any other media control (it is a simple Thinkpad keyboard). My headphone is a Parrot.

Here is the output, cleared out from irrelevant stuff:

```
➤  /usr/lib/x86_64-linux-gnu/cinnamon-settings-daemon/csd-media-keys --verbose
[...]
(csd-media-keys:13678): Cvc-DEBUG: 17:55:29.221: 	Property: 'device.intended_roles' = 'phone'
(csd-media-keys:13678): media-keys-plugin-DEBUG: 17:55:41.201: Creating proxy for for org.mpris.MediaPlayer2.rhythmbox
```
It does not seem to catch the key press. Nothing happens (in between several presses, I started Rhythmbox).

**Comment by mtwebster** [comment]
Is this a "Parrot button" (programmable under different oses) ?  I'm considering trying to duplicate this issue, but I want to buy correct hardware.

What is the output currently for:
```
gsettings get org.cinnamon.desktop.keybindings.media-keys pause
```

**Comment by phocean** [comment]
> Is this a "Parrot button" (programmable under different oses) ? I'm considering trying to duplicate this issue, but I want to buy correct hardware.

It is not really supposed to be programmable, I guess: ([here is the model](https://www.parrot.com/fr/audio/parrot-zik-3#parrot-zik-3-details)).
It just that I know it works in Gnome-Shell on the same machine.
Do not spend your money on it yet, I am sure it can get sorted. And, at least, let me contribute somehow first.

Here is the output:

```
➤  gsettings get org.cinnamon.desktop.keybindings.media-keys pause
['XF86AudioPause', 'AudioPlay']

~
➤  gsettings get org.cinnamon.desktop.keybindings.media-keys play 
@as []
```

I am going to look for a keyboard with media keys in the next days, so that at least I can test this stuff with similar keys.

**Comment by mtwebster** [comment]
It looks as though from the settings that that key isn't even bound to anything.  Try this:

```
gsettings set org.cinnamon.desktop.keybindings.media-keys play "['XF86AudioPlay']"
```
It looks as though the teaching mechanism may be failing here, reading "AudioPlay" instead of "XF86AudioPlay"

Regardless, this should have all worked out of the box, as the default for play *is* "XF86AudioPlay"

(that is all pending on whether or not this test works)

**Comment by phocean** [comment]
Well, it does not make any difference.

On the other side, I found a little eSynic bluetooth keyboard and it works perfectly.

No difference between the two in xev:

```
# keyboard
MappingNotify event, serial 37, synthetic NO, window 0x0,
    request MappingKeyboard, first_keycode 8, count 248

# headset
MappingNotify event, serial 38, synthetic NO, window 0x0,
    request MappingKeyboard, first_keycode 8, count 248
```

And only the keyboard is seen by csd-media-keys:

```
(csd-media-keys:12716): Cvc-DEBUG: 22:27:00.894: Updating sink: index=1 name='bluez_sink.A0_14_3D_D3_C8_21.a2dp_sink' description='usura V3.07' map='front-left,front-right'
```

Also attaching here the full output of csd-media-keys debug, as it has some usefull lines describing the headset profile.

[csd-media-keys.log](https://github.com/linuxmint/Cinnamon/files/2386070/csd-media-keys.log)

**Comment by Vahan86** [comment]
@phocean, is this still an issue in Cinnamon 4.4?

**Comment by dcolascione** [comment]
I'm seeing this issue in Cinnamon 4.4.8. It's the weirdest thing. I can't get either cinnamon or xbindkeys to react to the key -- even if I use xkb or xmodmap (I tried both) to remap XF86AudioPlay to some other (e.g., F12). But if I press F12 manually, each binding works fine. What on earth?!

**Comment by Garbaz** [comment]
Since this is the first result for "linux mint XF86AudioPlay" on google, I'll add this:
My issue was that my headset sends "XF86AudioPlay" and "XF86AudioPause" in alternation for some reason (or maybe depending on whether something is playing? Not sure, doesn't matter). And while the "XF86AudioPause" event worked as intended, pausing e.g. VLC playback (globally), the "XF86AudioPlay" did nothing.
**My solution was the following:**
`gsettings set org.cinnamon.desktop.keybindings.media-keys play "['AudioPlay','XF86AudioPlay']"`
with the previous (default) value having been `"['AudioPlay']"`.
So far so good. Interestingly in the `System Settings->Keyboard->Shortcuts` menu, this appeared as two entries called "AudioPlay", as seen here:
![image](https://user-images.githubusercontent.com/13696798/94308030-2fb77f00-ff76-11ea-8921-54fb36be7b6c.png)

The bizarre thing though is that when deleting the second entry (from the UI), everything still works. Even after restarting the computer. And `gsettings get org.cinnamon.desktop.keybindings.media-keys play` had the old initial (non-working) value of `['AudioPlay']`. I have absolutely no clue why this is the case, and since I can no longer reproduce the original non-working state, I unfortunately can not do any further testing.
I hope this at least provides a fix for users with the same issue, though sadly not much of an explanation.
